modules/nlu-testing/src/backend/api.ts

```diff
@@ -181,10 +195,27 @@ function results2CSV(tests: Test[], results: _.Dictionary<TestResult>) {
   return stringify(records)
 }
 
-function isRunningFromSources(): string | undefined {
+async function isRunningFromSources(bp: typeof sdk, botId: string): Promise<string | undefined> {
   try {
+    const botConfig = (await bp.bots.getBotById(botId)) as BPDS_BotConfig
+    const bpdsId = botConfig.bpdsId
     const sourceDirectory = path.resolve(process.PROJECT_LOCATION, '../..')
-    const latestResultsPath = path.resolve(sourceDirectory, `./modules/nlu-testing/latest-results.csv`)
+    const botTemplatesPath = path.resolve(sourceDirectory, `./modules/nlu-testing/src/bot-templates`)
+    const childDirs = fs.readdirSync(botTemplatesPath)
+
+    let botTemplateUnderTesting = ''
+    for (const template of childDirs) {
+      const templatePath = path.resolve(botTemplatesPath, template)
+      const configPath = path.resolve(templatePath, 'bot.config.json')
+      const configContent = fs.readFileSync(configPath, { encoding: 'utf8' })
+      const config = JSON.parse(configContent) as BPDS_BotConfig
+      if (config.bpdsId === bpdsId) {
+        botTemplateUnderTesting = templatePath
+        break
```

Comment:
why not use a while instead ... you're clearly using a for loop with a break which is basically a while statement

Reply:
yes!